cns/restserver/util.go

```diff
@@ -923,3 +923,32 @@ func (service *HTTPRestService) createNetworkContainers(createNetworkContainerRe
 		Message:    "",
 	}
 }
+
+func (service *HTTPRestService) setResponse(w http.ResponseWriter, returnCode types.ResponseCode, response interface{}) {
+	serviceErr := service.Listener.Encode(w, &response)
+	logger.Response(service.Name, response, returnCode, serviceErr)
+}
+
+func (service *HTTPRestService) readSupportedApisCache() []string {
+	service.RLock()
```

Comment:
it's not advisable to use defers when it's not needed. It comes with some perf penalty. 
https://lk4d4.darth.io/posts/defer/

Reply:
That _was_ true. Note that this linked blog is from 2014 which means it was written when the latest version of Go was <1.5.
If we fast-forward a little, even as far back as [Go 1.14 the performance gap was **significantly** smaller](https://rakyll.org/inlined-defers/).
We're all the way up to go1.19 now and defer efficiency has been continuously improved. IMO, this is not a good reason to _not_ use `defer` anymore, except in the most performance-critical paths. I doubt we have anything running with such tight timing requirements that we care, and certainly not in this codepath.